applications/FluidDynamicsApplication/custom_utilities/fluid_characteristic_numbers_utilities.cpp

```diff
@@ -0,0 +1,376 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+//
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "includes/cfd_variables.h"
+#include "utilities/element_size_calculator.h"
+#include "utilities/geometry_utilities.h"
+#include "utilities/parallel_utilities.h"
+
+// Application includes
+#include "fluid_dynamics_application_variables.h"
+#include "fluid_characteristic_numbers_utilities.h"
+
+
+namespace Kratos
+{
+
+    void FluidCharacteristicNumbersUtilities::CalculateLocalCFL(ModelPart& rModelPart)
+    {
+        KRATOS_TRY;
+
+        // Get the projected element size function according to the corresponding geometry
+        // Note that in here it is assumed that all the elements in the model part feature the same geometry
+        const auto& r_geom = rModelPart.ElementsBegin()->GetGeometry();
+        ElementSizeFunctionType minimum_h_func = FluidCharacteristicNumbersUtilities::GetMinimumElementSizeFunction(r_geom);
+
+        // Calculate the CFL number in each element
+        const double current_dt = rModelPart.GetProcessInfo().GetValue(DELTA_TIME);
+        block_for_each(rModelPart.Elements(), [&](Element& rElement){
+            const double element_cfl = FluidCharacteristicNumbersUtilities::CalculateElementCFL(rElement, minimum_h_func, current_dt);
+            rElement.SetValue(CFL_NUMBER, element_cfl);
+        });
+
+        KRATOS_CATCH("")
+    }
+
+    double FluidCharacteristicNumbersUtilities::CalculateElementCFL(
```

Comment:
```suggestion
    double FluidCharacteristicNumbersUtilities::CalculateElementCFLNumber(
```

more consistent with the other fcts

Reply:
As this is used in other places such as the `EstimateDtUtils` I'd prefer keeping the current name. We can introduce this change and the one you suggest above in a unique PR. Makes sense for you?